### ISSUE TITLE
Choose quality - works only for some links

### ISSUE THREAD

**Issue by Baku86** [nilaoda/N_m3u8DL-CLI#405]
Hello,

I have two mpd links. One allows me to choose quality in N_m3u8DL-CLI:
https://s3-dub-ww.cf.dash.row.aiv-cdn.net/dm/2$UoK8aLmgs0nowiEXPa67jAGAF2s~/fa85/f264/7234/435b-b7b2-1da444af07ba/8c05b27e-7f9f-4da0-aa90-c84d40277b3f_corrected.mpd?encoding=segmentBase&amznDtid=AOAGZA014O5RE

Second one automatically downloads best available quality:
https://r.dcs.redcdn.pl/dash/o2/tvnplayer/vod/21_400_28121_0002/SMOOTH_HD_PL/041bc738-1032-4263-a618-0f8aa56be61a/playlist.ism/playlist.mpd?indexMode=&dummyfile=

How can I choose quality/stream in second link?

**Comment by nilaoda** [comment]
you can only choose quality when mpd has 2 or more languages audio.

**Comment by Baku86** [comment]
True, it has one audio, but few video streams. Can you change code to allow quality choosing in that case?

**Comment by p2mgrup** [comment]
+1

**Comment by Baku86** [comment]
It doesn't necessarily have to be hardcoded. In fact checkbox option or even some manual change in config file would be more flexible.

**Comment by Baku86** [comment]
Any chance for this king of enhancement?

**Comment by erickyun** [comment]
> True, it has one audio, but few video streams. Can you change code to allow quality choosing in that case?

+1

**Comment by m0ck69** [comment]
up

**Comment by astravaganza** [comment]
+1
